pkg/service/credential/storage.go

```diff
@@ -346,6 +346,36 @@ func (cs *Storage) getCredential(ctx context.Context, id string, namespace strin
 	return &stored, nil
 }
 
+// GetCredentials gets all credentials stored with a prefix key
+// The method is greedy, meaning if multiple values are found...and some fail during processing, we will
+// return only the successful values and log an error for the failures.
+func (cs *Storage) GetCredentials(ctx context.Context) ([]StoredCredential, error) {
+	keys, err := cs.db.ReadAllKeys(ctx, credentialNamespace)
+	if err != nil {
+		return nil, sdkutil.LoggingErrorMsgf(err, "could not read credential storage")
+	}
+
+	var storedCreds []StoredCredential
+	for _, key := range keys {
+		credBytes, err := cs.db.Read(ctx, credentialNamespace, key)
+		if err != nil {
+			logrus.WithError(err).Errorf("could not read credential with key: %s", key)
```

Comment:
Is the intention to skip this? I think we should return an error.

Reply:
yeah as the comment for the method says this is greedy - we warn we couldn't get one but return what we have. in case of data corruption we wouldn't want to eliminate the good data

we may want to amend this to return both data and an error

Reply:
Either way is fine. We just need to document this so it's reflected in our API.